sdk/communication/communication-phone-numbers/test/internal/headers.spec.ts

```diff
@@ -79,11 +78,18 @@ describe("PhoneNumbersClient - headers", function() {
   });
 
   it("sets bearer authorization header with TokenCredential", async function(this: Context) {
-    if (!canCreateRecordedClientWithToken() || isPlaybackMode()) {
+    if (isPlaybackMode()) {
```

Comment:
Why are we skipping this test in the case of playback mode?

Reply:
honestly, I'm actually not sure! I'm going to guess something to do with the fake Token that playback mode creates. I'll have to run and test it out

Reply:
Thanks for pointing this out @JoshuaLai, turns out this was failing because of recorder and env vars, but this never actually needed to use those env vars as it should not have been using a real token.
I've found a couple other cases that should've been using mock token credentials as well and updated those accordingly